huntsman/utils/pyro/__init__.py

```diff
@@ -0,0 +1,155 @@
+    # the same as the default gateway's IP.
+    if len(addresses) > 1:
+        msg = 'Interface has more than 1 IP address. Filtering on 1st byte...'
+        if verbose:
+            print(msg)
+        if logger:
+            logger.debug(msg)
+        byte1, byte2 = gateway_IP.split('.')[0:2]
+        addresses = [address for address in addresses if address.split('.')[0] == byte1]
+        if len(addresses) > 1:
+            msg = 'Interface still has more then 1 IP address. Filtering on 2nd byte...'
+            if verbose:
+                print(msg)
+            if logger:
+                logger.debug(msg)
+            addresses = [address for address in addresses if address.split('.')[1] == byte2]
+
+    assert len(addresses) == 1
+    host = addresses[0]
+    msg = 'Using IP address {} on interface {}'.format(addresses[0], interface)
+    if verbose:
+        print(msg)
+    if logger:
+        logger.debug(msg)
+    return host
+
+
+def run_name_server(host=None, port=None, autoclean=0):
+    """
+    Runs a Pyro name server.
+
+    The name server must be running in order to use distributed cameras with POCS. The name server
+    should be started before starting camera servers or POCS.
+
+    Args:
+        host (str, optional): hostname/IP address to bind the name server to. If not given then
+            get_own_ip will be used to attempt to automatically determine the IP addresses of
+            the computer that the name server is being started on.
+        port (int, optional): port number to bind the name server to. If not given then the port
+            will be selected automatically (usually 9090).
+        autoclean (int, optional): interval, in seconds, for automatic deregistration of objects
+            from the name server if they cannot be connected. If not given no autocleaning will
+            be done.
+    """
+    try:
+        # Check that there isn't a name server already running
+        name_server = Pyro4.locateNS()
+    except errors.NamingError:
+        if not host:
+            # Not given an hostname or IP address. Will attempt to work it out.
+            host = get_own_ip(verbose=True)
+        else:
+            host = str(host)
+
+        if port:
+            port = int(port)
+        else:
+            port = None
+
+        Pyro4.config.NS_AUTOCLEAN = float(autoclean)
+
+        print("Starting Pyro name server... (Control-C/Command-C to exit)")
+        naming.startNSloop(host=host, port=port)
+    else:
+        print("Pyro name server {} already running! Exiting...".format(name_server))
+
+
+def run_camera_server(ignore_local):
+    """
+    Runs a Pyro camera server.
+
+    The camera server should be run on the camera control computers of distributed cameras. The
+    camera servers should be started after the name server, but before POCS.
+
+    Args:
+        ignore_local (bool, optional): If True use the default $POCS/conf_files/pyro_camera.yaml
```

Comment:
s/$POCS/$HUNTSMAN_POCS/

Reply:
Yes.

Reply:
I don't see this change.

Reply:
Fixed another instance of the same thing and missed this one!